daemon/cmd/devices.go

```diff
@@ -97,8 +97,8 @@ func (dm *DeviceManager) Detect() error {
 		}
 	}
 
-	detectDirectRoutingDev := option.Config.EnableNodePort
-	if option.Config.DirectRoutingDevice != "" {
+	detectDirectRoutingDev := option.Config.DirectRoutingDeviceRequired()
+	if detectDirectRoutingDev && option.Config.DirectRoutingDevice != "" {
```

Comment:
I think it would be a bit easier to parse without the `detectDirectRoutingDev` variable.

Reply:
Sorry, but to be clarify are you saying that we should read the `option.Config.DirectRoutingDevice` even if the Direct Routing Device is not required? Or, saying we should be like this?

```
detectDirectRoutingDev := option.Config.DirectRoutingDeviceRequired()
needDirectRoutingDev := detectDirectRoutingDev
if needDirectRoutingDev && option.Config.DirectRoutingDevice != "" {
```

Reply:
I'm saying it could be like:
```suggestion
	if option.Config.DirectRoutingDeviceRequired() &&
		option.Config.DirectRoutingDevice != "" {
```